lib/ably/realtime/channel.rb

```diff
@@ -164,7 +164,12 @@ def publish(name, data = nil, attributes = {}, &success_block)
           [{ name: name, data: data }.merge(attributes)]
         end
 
-        queue_messages(messages).tap do |deferrable|
+        if messages.length > Realtime::Connection::MAX_PROTOCOL_MESSAGE_BATCH_SIZE
+          error = Ably::Exceptions::InvalidRequest.new("It is not possible to publish more than #{Realtime::Connection::MAX_PROTOCOL_MESSAGE_BATCH_SIZE} messages with a single publish request.")
+          return Ably::Util::SafeDeferrable.new_and_fail_immediately(logger, error)
+        end
```

Comment:
Where'd this MAX_PROTOCOL_MESSAGE_BATCH_SIZE requirement come from? Don't remember any discussion on this, and it's not in the spec afaict

Reply:
See https://github.com/ably/ably-ruby/pull/166/commits/fd267e6599e47f25eb81551924f5322a321ce386.  I mentioned in that commit that this limit existed before, so I have kept it as it was.  @paddybyers I believe we have a max `ProtocolMessage` size, but do we have / want any limit on how many messages can exist within a single `ProtocolMessage`?